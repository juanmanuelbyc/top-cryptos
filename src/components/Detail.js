import PropTypes from 'prop-types';

const Detail = (props) => {
  const { title, value } = props;

  return (
    <div className="detailCard">
      <h3 className="detailCardText">{title}</h3>
      <h3 className="detailCardText">{value}</h3>
    </div>
  );
};

Detail.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
};

export default Detail;
