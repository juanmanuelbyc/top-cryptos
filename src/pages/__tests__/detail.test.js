import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Detail from '../../components/detail';

const TestDetail = () => {
  const history = createMemoryHistory();
  return (
    <Router location={history.location} navigator={history}>
      <Detail title="Price" value="99999" />
    </Router>
  );
};

it('renders correctly', () => {
  const tree = renderer.create(
    <TestDetail />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
