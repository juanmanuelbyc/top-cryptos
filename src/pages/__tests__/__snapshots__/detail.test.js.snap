// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="detailCard"
>
  <h3
    className="detailCardText"
  >
    Price
  </h3>
  <h3
    className="detailCardText"
  >
    99999
  </h3>
</div>
`;
